### PR TITLE
Adds ability to specify specific relations to be included

### PR DESCRIPTION
**Pull request by jamesdixon**
@chamini2 as you suggested previously, this PR allows for specific relations to be included. I already ran into a case for this, hence the quick addition 👍

### CODE REVIEW

**Comment by chamini2** [comment]
Hey, I don't think this is the best API to achieve this.

I was thinking maybe then to do:

```js
relations : false // no relations, none included
relations : true | relations : { included: true } | relations : {} // all relations, all included
relations : { included: false } // all relations, none included
relations : { included: { a: true, b: false } } // relations: 'a' 'b', included: 'a'
relations : { included: ['a', 'b'] } } // relations and included: 'a' 'b'
```
```js
relations : false // no relations, none included
relations : true | relations : { included: true } | relations : {} // all relations, all included
relations : { included: false } // all relations, none included
relations : ['a', 'b'] // relations and included: 'a' 'b'
relations : { 'a': true, 'b': false } // relations: 'a' 'b', included: 'a'
```

So, dropping entirely the `fields` option?

I don't know if it's the best, but it limits problems in input. With the proposed API by this PR think of this case:

```js
relations : { fields: ['a', 'b'], included: ['a', 'c'] }
```

**Comment by jamesdixon** [comment]
Hey. I appreciated the suggestion, but I believe the API you proposed is too complex. It just doesn't read well and ultimately seems more of a pain to implement.

However, I do agree that the case you mentioned is problematic. I think we could easily difference those arrays to make it work. Essentially, if `fields` exists and `included` is an array, we difference on those to get the fields that can actually be included.

What do you think?

**Comment by jamesdixon** [comment]
```
relations : { included: {a: true, b: false} } // relations: 'a' 'b', included: 'a'
```

You still run into the original issue of not being able to specify the relations to serialize, correct?

For example, without the `fields` param, the above only allows you to serialize all relations and selectively include or exclude them from the payload.

**Comment by chamini2** [comment]
> For example, without the fields param, the above only allows you to serialize all relations and selectively include or exclude them from the payload.

I was thinking that the properties of that object would be the ones to add in relations and just include the `true` ones in the `included` array. But I see your point in it being complex.

> However, I do agree that the case you mentioned is problematic. I think we could easily difference those arrays to make it work. Essentially, if fields exists and included is an array, we difference on those to get the fields that can actually be included.

I guess this works, I don't love it but seems easier to understand.

**Comment by jamesdixon** [comment]
@chamini2 see the changes I just pushed.

> I guess this works, I don't love it but seems easier to understand.

It seems like a valid solution; not sure how we'd make it much better without making the API over-complex. What don't you love about it? ❤️  😄

**Comment by jamesdixon** [comment]
Ultimately, the solution I provided is just eliminating the ability for a user to make an error. I believe we're going to encounter that regardless of what the API looks like.

File: src/bookshelf/utils.ts [hunk 1]
```diff
@@ -130,9 +132,34 @@ function relationAllowed(bookOpts: BookOpts, relName: string): boolean {
...
+  }
+}
+
+/**
+ * Based on Bookshelf options, determine if a relation must be included
+ */
+function includeAllowed(bookOpts: BookOpts, relName: string): boolean {
+  let { relations }: BookOpts = bookOpts;
+  let { fields }: RelationOpts = relations;
+  let { included }: RelationOpts = relations;
```
Comment (chamini2) [inline src/bookshelf/utils.ts]: You could change this to:

```js
let { fields, included }: RelationOpts = relations;
```

Comment (jamesdixon) [inline src/bookshelf/utils.ts]: Sorry. Will do. Just a lot of copying pasting

**Comment by jamesdixon** [comment]
@chamini2 just added the test from the other branch and replaced `some` with `includes` in the `relationsAllowed` block.

**Comment by chamini2** [comment]
Hey @jamesdixon, I opened a PR to this branch in your repo to change a bit the checks for options passed. I figured it would be easier to change it myself.

**Comment by jamesdixon** [comment]
Ok, changes are fine by me.

**Comment by jamesdixon** [comment]
Thanks for your help!

**Pull request merged**